sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala

```diff
@@ -137,6 +136,38 @@ object CommandUtils extends Logging {
     size
   }
 
+  def calculateLocationsSizes(
+      sparkSession: SparkSession,
+      tid: TableIdentifier,
+      paths: Seq[Option[URI]]): Long = {
+    if (sparkSession.sessionState.conf.parallelFileListingInStatsComputation) {
+      calculateLocationsSizesParallel(sparkSession, paths.map(_.map(new Path(_))))
+    } else {
+      paths.map(p => calculateLocationSize(sparkSession.sessionState, tid, p)).sum
+    }
+  }
+
+  /**
+   * Launch a Job to list all leaf files in `paths` and compute the total size
+   * for each path.
+   * @param sparkSession the [[SparkSession]]
+   * @param paths the Seq of [[Option[Path]]s
+   * @return total size of all partitions
+   */
+  def calculateLocationsSizesParallel(
```

Comment:
ditto. `Locations` and `Sizes`.

Reply:
Renamed to `calculateLocationSizeParallel`, done in https://github.com/apache/spark/pull/27413/commits/ac8b771c1264fc8069a61a3cdb338f4ac6c406af, please check.